addon/addon/modifiers/sprite.ts

```diff
@@ -11,23 +12,34 @@ interface SpriteModifierArgs {
 }
 
 export interface SpriteModifierSignature {
-  Element: HTMLElement;
+  Element: Element;
   Args: SpriteModifierArgs;
 }
 
+function cleanup(instance: SpriteModifier): void {
+  instance.animations.unregisterSpriteModifier(instance);
+}
+
 export default class SpriteModifier extends Modifier<SpriteModifierSignature> {
+  element!: Element;
   id!: string;
   role: string | null = null;
 
   @service declare animations: AnimationsService;
 
-  didReceiveArguments(): void {
-    this.id = this.args.named.id;
-    this.role = this.args.named.role ?? null;
-    this.animations.registerSpriteModifier(this);
+  constructor(owner: unknown, args: ArgsFor<SpriteModifierSignature>) {
+    super(owner, args);
+    registerDestructor(this, cleanup);
   }
 
-  willRemove(): void {
-    this.animations.unregisterSpriteModifier(this);
+  modify(
+    element: Element,
+    _: PositionalArgs<SpriteModifierSignature>,
+    { id, role }: NamedArgs<SpriteModifierSignature>
+  ): void {
+    this.element = element;
+    this.id = id;
+    this.role = role ?? null;
+    this.animations.registerSpriteModifier(this);
```

Comment:
Argument changes trigger a registerSpriteModifier call for an already registered sprite modifier. We can handle this in the animations service or the AnimationParticipantManager, but should we gate it here?

Reply:
I think it's better to prevent calling `registerSpriteModifier` when arguments change, from within this modifier. We could do both though.

Reply:
PR: https://github.com/cardstack/boxel-motion/pull/137